cellmlmanip/model.py

```diff
@@ -631,7 +631,7 @@ def graph_with_sympy_numbers(self):
             # And replace the equation with one with the rhs subbed with sympy.Number objects
             if subs_dict:
                 # Update rhs
-                rhs = equation.rhs.subs(subs_dict)
```

Comment:
This does change behaviour, since `subs` will simplify whereas `xreplace` will not. So if e.g. we get `0 * expr` after substitution then with `subs` we could remove dependencies on other variables, hence the need for the following lines removing edges.

Possibly that's ok, but it definitely needs discussion. And we should also check whether the following lines can indeed be removed with this change.

Reply:
Actually xreplace will also simplify or at least it did in my test cases, despite the docs not mentioning it.

Reply:
OK, maybe just more limited canonicalisation? Not sure! I'm surprised that the apparently more complex `replace` would be faster.

Could also consider the `simultaneous=True` argument to `subs`.

Reply:
yeah, looking at the code for `xreplace` it is just recursively doing `self.func(*args)` after calling xreplace on its arguments, so the only simplification is that done by construction.

Reply:
Looking at the code for subs, it will call `sympify` explicitly on both old & new items, which will be a performance hit. It also does a lot more complex recursion down the tree than does `xreplace`.

`replace` will also `_sympify` its arguments and do more complex recursion.

Reply:
Ok I reverted to subs here